### ISSUE TITLE
different version different error

### ISSUE THREAD

**Issue by Fatihlrcfs** [wdecoster/methplotlib#36]
Hi @wdecoster,

Firstly, thank you for providing this program. I am currently working on one of nanopolish results and ı have a methylation call and frequency .tsv file. The methplotlib version 0.8 works fine on that input and generates graphs, however, a newer version of methplotlib like 0.17 and 0.20 gives an error. I have the same environment for them but get different errors.  Many thanks. 

2022-04-19 10:20:56,900 methplotlib 0.17.0 started.
2022-04-19 10:20:56,901 Python version is: 3.9.12 (main, Apr  5 2022, 06:56:58)  [GCC 7.5.0]
2022-04-19 10:20:56,901 Arguments are: Namespace(methylation=['61fast28x2_methylation_calls.tsv', '61fast28x2_methylation_frequency.tsv'], names=['calls', '61fast28x2testmethplotlib17'], window='16:10,980,759-11,045,000', gtf='GRCh38_latest_genomic.gff.gz', bed='61fast28x2.sorted.bed', fasta=None, simplify=True, split=False, static=None, binary=False, smooth=5, dotsize=4, example=False, outfile=None, qcfile=None, store=False)
2022-04-19 10:20:56,901 Processing 16_10980759_11045000
2022-04-19 10:20:56,937 File 61fast28x2_methylation_calls.tsv is of type nanopolish_call
2022-04-19 10:20:56,937 Reading 61fast28x2_methylation_calls.tsv slowly by splitting the file in chunks.
2022-04-19 10:23:11,798 Read the file in a dataframe.
2022-04-19 10:23:11,825 Error processing 61fast28x2_methylation_calls.tsv.
2022-04-19 10:23:11,825 "None of [Index(['Start', 'End'], dtype='object')] are in the [columns]"
Traceback (most recent call last):
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/R/lib/python3.9/site-packages/methplotlib/import_methylation.py", line 47, in read_meth
    return parse_nanopolish(filename, file_type, name, window, smoothen=smoothen)
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/R/lib/python3.9/site-packages/methplotlib/import_methylation.py", line 101, in parse_nanopolish
    gr.pos = np.floor(gr.drop().df[["Start", "End"]].mean(axis=1))
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/R/lib/python3.9/site-packages/pandas/core/frame.py", line 3511, in __getitem__
    indexer = self.columns._get_indexer_strict(key, "columns")[1]
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/R/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 5782, in _get_indexer_strict
    self._raise_if_missing(keyarr, indexer, axis_name)
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/R/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 5842, in _raise_if_missing
    raise KeyError(f"None of [{key}] are in the [{axis_name}]")
KeyError: "None of [Index(['Start', 'End'], dtype='object')] are in the [columns]"

and error for version 0.20
Reading 61fast28x2_methylation_calls.tsv would be faster with bgzip and tabix.
Please index with 'tabix -S1 -s1 -b3 -e4'.
/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/import_methylation.py:50: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.
  return parse_nanopolish(filename, file_type, name, window, smoothen=smoothen)

Reading 61fast28x2_methylation_frequency.tsv would be faster with bgzip and tabix.
Please index with 'tabix -S1 -s1 -b2 -e3'.
/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/annotation.py:146: DtypeWarning:

Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.

Traceback (most recent call last):
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/bin/methplotlib", line 10, in <module>
    sys.exit(main())
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/methplotlib.py", line 36, in main
    minqual=args.minqual,  # only for input in bam/cram format
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/methplotlib.py", line 107, in meth_browser
    annotation_traces, y_max = plots.gtf_annotation(gtf, window, simplify)
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/plots.py", line 38, in gtf_annotation
    annotation = parse_annotation(gtf, window, simplify)
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/annotation.py", line 92, in parse_annotation
    for line in open_gtf(gtff) if good_record(line, window.chromosome)],
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/annotation.py", line 92, in <listcomp>
    for line in open_gtf(gtff) if good_record(line, window.chromosome)],
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/annotation.py", line 35, in good_record
    if line.startswith(chromosome) \
TypeError: startswith first arg must be str or a tuple of str, not int

**Comment by wdecoster** [comment]
Can you check that the gff file you are using uses "16" as a chromosome identifier?

**Comment by wdecoster** [comment]
Oops - didn't mean to close :)

**Comment by Fatihlrcfs** [comment]
Hi @wdecoster ,

thanks for the quick response.  I have an issue with my cluster now but ı wanna say that this .gff file works fine with version 0.8. Also, I have different gff3, gff and gtf. files and all dont work for both versions (0.17 and 0.20). many thanks.

best.

**Comment by wdecoster** [comment]
I just pushed a new version to PyPI, v0.20.1, can you see if that solves anything?
I should really rewrite the GFF/GTF parsing code but I have no time.

**Comment by Fatihlrcfs** [comment]
Hi @wdecoster, 

sorry for the late return. now its works with nanopolish output. I am also working with the megalodon output and ı couldn't  able to run the cram file in methplotlib and ı am getting the error below. do you have any help with that? thanks for your help.  

The log file  
2022-04-20 12:01:28,810 methplotlib 0.20.1 started.
2022-04-20 12:01:28,810 Python version is: 3.7.13 (default, Mar 29 2022, 02:18:16)  [GCC 7.5.0]
2022-04-20 12:01:28,810 Arguments are: Namespace(bed='modified_bases.5mC.bed', binary=False, dotsize=4, example=False, fasta=None, gtf='GRCh38genomic.gff', methylation=['mapping_sorted.cram'], minqual=20, names=['calls'], outfile=None, qcfile=None, simplify=False, smooth=5, split=False, static=None, store=False, window='16:10,000,000-12,000,000')
2022-04-20 12:01:28,813 Processing 16_10000000_11000000
2022-04-20 12:01:28,822 File mapping_sorted.cram is of type bam
2022-04-20 12:01:28,995 Collected methylation data for 0 datasets
2022-04-20 12:01:29,575 Created QC plots
2022-04-20 12:01:29,575 Prepared methylation traces.
2022-04-20 12:01:29,575 Making browser in overlaying mode.
 
The error:
Traceback (most recent call last):
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/bin/methplotlib", line 10, in <module>
    sys.exit(main())
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/methplotlib.py", line 36, in main
    minqual=args.minqual,  # only for input in bam/cram format
  File "/cluster/lrcfs/ftiras/ftiras_env/envs/Methplotlib17/lib/python3.7/site-packages/methplotlib/methplotlib.py", line 94, in meth_browser
    if meth_traces.types[0] == 'nanopolish_freq':
IndexError: list index out of range 

many thanks. great days,

Fatih.

**Comment by wdecoster** [comment]
What is the output of the following?

`samtools view mapping_sorted.cram 16:10,000,000-12,000,000 | wc -l`

**Comment by Fatihlrcfs** [comment]
hi @wdecoster ,

thanks for your answer. ı got this after typing your advice. I think something wrong/ not ptoperly set up my enviroment. many thanks. best wishes 
samtools: /cluster/lrcfs/2397405/ftiras_env/bin/../lib/libtinfow.so.6: no version information available (required by samtools)
samtools: /cluster/lrcfs/2397405/ftiras_env/bin/../lib/libncursesw.so.6: no version information available (required by samtools)
samtools: /cluster/lrcfs/2397405/ftiras_env/bin/../lib/libncursesw.so.6: no version information available (required by samtools)

**Comment by wdecoster** [comment]
Do you think you can share that file?

**Comment by Fatihlrcfs** [comment]
hi @wdecoster  yes I can but the github dont support to upload the file do you know how can ı upload it ? thanks.

**Comment by wdecoster** [comment]
How large is the file in megabytes or gigabytes?

**Comment by Fatihlrcfs** [comment]
252.630 kb

**Comment by wdecoster** [comment]
Okay, can you try uploading to https://www.dropbox.com/request/lfyAHhYE5nslcjMWmgc2 ?

**Comment by Fatihlrcfs** [comment]
Hi @wdecoster ,

thanks for sending link. I uploaded the mapping_sorted.cram file

**Comment by wdecoster** [comment]
I don't see any modified nucleotides in this file. The information simply isn't there. How was this file generated?

**Comment by Fatihlrcfs** [comment]
Hi @wdecoster , 

first, we used megalodon for generating cram file. then samtools was used for sorted and indexing. Also, we generated some data from similar data with nanopolish package. thanks.



megalodon command: "qsub -pe smp 40 -jc long -M 2397405@dundee.ac.uk -m ae -cwd -b y -R y megalodon Firstrun61fast5subset/ --reference wholegenome.fasta --outputs basecalls mappings mods --mappings-format cram --sort-mappings --processes 40 --guppy-server-path /cluster/lrcfs/2397405/bin/ont-guppy-cpu-6.0.1/bin/guppy_basecall_server --guppy-params "-d /cluster/lrcfs/2397405/bin/rerio/basecall_models/ --num_callers 40" --guppy-timeout 400 --guppy-config res_dna_r941_min_modbases_5mC_CpG_v001.cfg --overwrite --output-directory megalodon_results_secondcram61seconda "

samtools command:"  samtools index mappings.cram
  samtools sort mappings.cram > mapping_sorted.cram
  samtools index mapping_sorted.cram"
